### DISCUSSION TITLE
How to find data according to their relationships count like User where posts.length === 0

### DISCUSSION THREAD

**Discussion by ragokan** [prisma/prisma#11983] [Q&A] [answered]
Hello, I want to find data according to the relationship counts, but currently I am not able to do it. A basic example

```ts
const usersWithoutPost = await prisma.user.findMany({ where: { posts: { length: 0 } } })
```

But I can only do every, none and some. So instead I do this:
```ts
const allUsers = await prisma.user.findMany({ include: { posts: true } );
const usersWithoutPost = allUsers.filter((user) => user.posts.length === 0);
```

**Comment by austincrim** [comment, marked as answer]
Hey @ragokan 👋 ,

I believe you are looking for [this type of query](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#get-all-user-records-with-zero-posts):
```ts
const usersWithoutPost = await prisma.user.findMany({ where: { posts: { none: {} } } })
```
From our docs:
> You can use `none` without parameters to return all records with no relations

Let me know if that works!

**Reply by ragokan** [reply]
It did work! Thank you so much @austincrim. It was my bad to not to read the docs correctly.

**Reply by boredland** [reply]
Is there a variant where I can search for a specific length? This only allows me to search for an empty relation, doesn't it...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
